src/utils/tracked-controls.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Return enumerated gamepads matching id prefix.
+ *
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ */
+module.exports.getGamepadsByPrefix = function (idPrefix) {
+  var rtn = [];
+  var gamepads = navigator.getGamepads && navigator.getGamepads();
+  if (gamepads) {
+    for (var i = 0; i < gamepads.length; ++i) {
+      var gamepad = gamepads[i];
+      if (gamepad) {
```

Comment:
Is this check necessary? If you are iterating over the array there should be a gamepad.